### PR TITLE
OCPBUGS#16663: Remove provisioning with `new-master-machine.yaml`

### PR DESCRIPTION
**Pull request by kowen-rh**
Version(s):
4.12+

Issue:
[OCPBUGS-16663](https://issues.redhat.com/browse/OCPBUGS-16663)

Link to docs preview:
[Restoring to a previous cluster state](https://80267--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state.html#dr-scenario-2-restoring-cluster-state_dr-restoring-cluster-state)
[Replacing an unhealthy etcd member whose machine is not running or whose node is not ready](https://80267--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/replacing-unhealthy-etcd-member.html#restore-replace-stopped-etcd-member_replacing-unhealthy-etcd-member)
[Replacing an unhealthy bare metal etcd member whose machine is not running or whose node is not ready](https://80267--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/replacing-unhealthy-etcd-member.html#restore-replace-stopped-baremetal-etcd-member_replacing-unhealthy-etcd-member)

QE review:
- [x] QE has approved this change.

Additional information:
N/A

### CODE REVIEW

File: modules/dr-restoring-cluster-state.adoc
Comment (dusk125) [inline modules/dr-restoring-cluster-state.adoc:472]: Since the machine is deleted and automatically provisioned almost immediately, this can be removed in favor of just verifying that the machine is in the provisioning state.

**Review by dusk125 — CHANGES_REQUESTED** [review]
Otherwise looks good to me and works a treat!

File: modules/dr-restoring-cluster-state.adoc
Comment (kcarmichael08) [inline modules/dr-restoring-cluster-state.adoc:478]:
```suggestion
.. Verify that a new machine has been created by running the following command:
```
Suggestion per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command - but if that would make it inconsistent with the phrasing in the other steps, it might be best to leave alone. Your call.

**Review by kcarmichael08 — COMMENTED** [review]
One suggestion; otherwise looks great!

File: modules/dr-restoring-cluster-state.adoc
Comment (kowen-rh) [reply modules/dr-restoring-cluster-state.adoc:478]: I noticed that while working on this, too, but kept it for consistency. I might create a new issue to change that phrasing throughout this module.

Thanks!

File: modules/restore-replace-stopped-baremetal-etcd-member.adoc
Comment (mburke5678) [inline modules/restore-replace-stopped-baremetal-etcd-member.adoc:201]: Probably not worth holding up this PR, but after removing this substep, the [previous substep](https://github.com/openshift/openshift-docs/pull/80267/files#diff-33208a1472920fee85adcb90269b5d140d0d2c19752c3ed7bc2412df2774e047R180) should become a unnumbered step (i.e. with an asterisk) as it is the only substep. 
 
.. Obtain the machine for the unhealthy member.

Comment (mburke5678) [reply modules/restore-replace-stopped-baremetal-etcd-member.adoc:201]: Actually, the wording in [step 4](https://github.com/openshift/openshift-docs/pull/80267/files#diff-33208a1472920fee85adcb90269b5d140d0d2c19752c3ed7bc2412df2774e047R176-R178) doesn't make sense after you remove the substeps, as all you are doing is listing nodes to identify the unhealthy member.

Comment (mburke5678) [reply modules/restore-replace-stopped-baremetal-etcd-member.adoc:201]: @kowen-rh PTAL ^^^^

Comment (kowen-rh) [reply modules/restore-replace-stopped-baremetal-etcd-member.adoc:201]: Nice catch, thank you!

Comment (kowen-rh) [reply modules/restore-replace-stopped-baremetal-etcd-member.adoc:201]: @mburke5678 I ended up reorganizing this slightly so that the previous substep is now its own step. This makes sense to me and reads well IMO.

**Pull request merged**
